src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -0,0 +1,282 @@
+                    if (baseTypes.Contains(controllerTypeSymbol) ||
+                        baseTypes.Contains(controllerBaseTypeSymbol))
+                    {
+                        // The controller class is protected by a validate anti forgery token attribute
+                        if (controllerTypeSymbol.GetAttributes().Any(s => s_AntiForgeryAttributeRegex.IsMatch(s.AttributeClass.Name)))
+                        {
+                            usingValidateAntiForgeryAttribute = true;
+
+                            return;
+                        }
+
+                        foreach (var actionMethodSymbol in controllerTypeSymbol.GetMembers().OfType<IMethodSymbol>())
+                        {
+                            // The method is protected by a validate anti forgery token attribute
+                            if (actionMethodSymbol.GetAttributes().Any(s => s_AntiForgeryAttributeRegex.IsMatch(s.AttributeClass.Name)))
+                            {
+                                usingValidateAntiForgeryAttribute = true;
+
+                                return;
+                            }
+
+                            if (actionMethodSymbol.IsPublic() &&
+                                !actionMethodSymbol.IsStatic &&
+                                !actionMethodSymbol.HasAttribute(nonActionAttributeTypeSymbol))
+                            {
+                                if (actionMethodSymbol.HasAttribute(httpPostAttributeTypeSymbol))
+                                {
+                                    actionMethodSymbols.Add((actionMethodSymbol, "HttpPost"));
+                                }
+                                else if (actionMethodSymbol.HasAttribute(httpPutAttributeTypeSymbol))
+                                {
+                                    actionMethodSymbols.Add((actionMethodSymbol, "HttpPut"));
+                                }
+                                else if (actionMethodSymbol.HasAttribute(httpDeleteAttributeTypeSymbol))
+                                {
+                                    actionMethodSymbols.Add((actionMethodSymbol, "HttpDelete"));
+                                }
+                                else if (actionMethodSymbol.HasAttribute(httpPatchAttributeTypeSymbol))
+                                {
+                                    actionMethodSymbols.Add((actionMethodSymbol, "HttpPatch"));
+                                }
+                            }
+                        }
+                    }
+                }, SymbolKind.NamedType);
+
+                compilationStartAnalysisContext.RegisterCompilationEndAction(
+                (CompilationAnalysisContext compilationAnalysisContext) =>
+                {
+                    if (usingValidateAntiForgeryAttribute && !hasGlobalAntiForgeryFilter && actionMethodSymbols.Count != 0)
```

Comment:
>!hasGlobalAntiForgeryFilter [](start = 61, length = 27)

Since we won't produce diagnostics if hasGlobalAntiForgeryToken becomes true, we could bail out in all the other callbacks if hasGlobalAntiForgeryToken is true.